packages/backend/src/tokens/authObjects.ts

```diff
@@ -0,0 +1,139 @@
+import type { ActClaim, JwtPayload, ServerGetToken, ServerGetTokenOptions } from '@clerk/types';
+
+import { type Organization, type Session, type User, createBackendApiClient } from '../api';
+
+export type SignedInAuthObjectOptions = {
+  apiKey: string;
+  apiUrl: string;
+  apiVersion: string;
+  token: string;
+  session?: Session;
+  user?: User;
+  organization?: Organization;
+};
+
+export type SignedInAuthObject = {
+  sessionClaims: JwtPayload;
+  sessionId: string;
+  session: Session | undefined;
+  actor: ActClaim | undefined;
+  userId: string;
+  user: User | undefined;
+  orgId: string | undefined;
+  orgRole: string | undefined;
+  orgSlug: string | undefined;
+  organization: Organization | undefined;
+  getToken: ServerGetToken;
+};
+
+export type SignedOutAuthObject = {
+  sessionClaims: null;
+  sessionId: null;
+  session: null;
+  actor: null;
+  userId: null;
+  user: null;
+  orgId: null;
+  orgRole: null;
+  orgSlug: null;
+  organization: null;
+  getToken: ServerGetToken;
+};
+
+export type AuthObject = SignedInAuthObject | SignedOutAuthObject;
+
+export function signedInAuthObject(sessionClaims: JwtPayload, options: SignedInAuthObjectOptions): SignedInAuthObject {
+  const {
+    act: actor,
+    sid: sessionId,
+    org_id: orgId,
+    org_role: orgRole,
+    org_slug: orgSlug,
+    sub: userId,
+  } = sessionClaims;
+  const { apiKey, apiUrl, apiVersion, token, session, user, organization } = options;
+
+  const { sessions } = createBackendApiClient({
+    apiKey,
+    apiUrl,
+    apiVersion,
+  });
+
+  const getToken = createGetToken({
+    sessionId,
+    sessionToken: token,
+    fetcher: (...args) => sessions.getToken(...args),
+  });
+
+  return {
+    actor,
+    sessionClaims,
+    sessionId,
+    session,
+    userId,
+    user,
+    orgId,
+    orgRole,
+    orgSlug,
+    organization,
+    getToken,
+  };
+}
+
+export function signedOutAuthObject(): SignedOutAuthObject {
+  return {
+    sessionClaims: null,
+    sessionId: null,
+    session: null,
+    userId: null,
+    user: null,
+    actor: null,
+    orgId: null,
+    orgRole: null,
+    orgSlug: null,
+    organization: null,
+    getToken: () => Promise.resolve(null),
+  };
+}
+
+export function prunePrivateMetadata(resource?: { private_metadata: any } | { privateMetadata: any } | null) {
```

Comment:
I feel that this helper should be an instance method in a future iteration of the backend resources. Eg `resource.toSSRJson` or similar, otherwise it might be easy to forget pruning important private data

Reply:
I agree although I am not sure if we should pursuit MVC for our backend resources. Let's defer the conversation.